src/main/java/daomephsta/unpick/impl/AbstractInsnNodes.java

```diff
@@ -58,9 +58,25 @@ public static boolean isLiteral(AbstractInsnNode insn, Object literal)
 			Character charLiteral = (char) (int) literalValue;
 			return literal.equals(charLiteral);
 		}
-		// Compare numbers by long value, to support widening comparisons
-		if (literalValue instanceof Number && literal instanceof Number)
+		// Compare integers by long value, to support widening comparisons
+		if (isIntegral(literalValue) && isIntegral(literal))
 		    return ((Number) literalValue).longValue() == ((Number) literal).longValue();
+		// Compare floating point numbers by double value, to support widening comparisons
+		if (isFloatingPoint(literalValue) && isFloatingPoint(literal))
```

Comment:
```suggestion
		if (isFloatingPoint(literalValue) && literal instanceof Number)
```
Think this is the right way in case you use an int constant to get a value for a double variable/parameter

Reply:
I'd rather deal with #8 on its own, rather than trying to guess what's needed before Unpick has tests for conversions.